4.1rc1: 1 test failure on openSUSE Tumbleweed i586 (32-bit)

[OpenSUSE Tumbleweed i586 on OBS](https://build.opensuse.org/package/show/home:bnavigator:branches:devel:languages:python:numeric/python-astropy-rc?rev=7), all bundled libs except for erfa:

[python-astropy-4.1rc1_test_i586_log.txt](https://github.com/astropy/astropy/files/4886334/python-astropy-4.1rc1_test_i586_log.txt):
```
[  274s] + PYTHONPATH=/home/abuild/rpmbuild/BUILD/astropy-4.1rc1:/home/abuild/rpmbuild/BUILDROOT/python-astropy-4.1.0~rc1-39.1.i386/usr/lib/python3.8/site-packages
[  274s] + PYTHONDONTWRITEBYTECODE=1
[  274s] + pytest-3.8 --ignore=_build.python2 --ignore=_build.python3 --ignore=_build.pypy3 -v --ignore docs/whatsnew
[  275s] Internet access disabled
[  275s] ============================= test session starts ==============================
[  275s] platform linux -- Python 3.8.3, pytest-5.4.3, py-1.9.0, pluggy-0.13.1 -- /usr/bin/python3
[  275s] cachedir: .pytest_cache
[  275s] hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/home/abuild/rpmbuild/BUILD/astropy-4.1rc1/.hypothesis/examples')
[  275s] 
[  275s] Running tests with Astropy version 4.1rc1.
[  275s] Running tests in astropy docs.
[  275s] 
[  275s] Date: 2020-07-07T17:26:22
[  275s] 
[  275s] Platform: Linux-5.7.5-1-pae-i686-athlon-with-glibc2.1.3
[  275s] 
[  275s] Executable: /usr/bin/python3
[  275s] 
[  275s] Full Python Version: 
[  275s] 3.8.3 (default, May 17 2020, 14:48:56) [GCC]
[  275s] 
[  275s] encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
[  275s] byteorder: little
[  275s] float info: dig: 15, mant_dig: 15
[  275s] 
[  275s] Package versions: 
[  275s] Numpy: 1.18.4
[  275s] Scipy: 1.4.1
[  275s] Matplotlib: 3.2.2
[  275s] h5py: 2.10.0
[  275s] Pandas: 1.0.5
[  275s] PyERFA: 4.1rc1
[  275s] 
[  275s] Using Astropy options: remote_data: none.
[  275s] 
[  275s] Matplotlib: 3.2.2
[  275s] Freetype: 2.10.2
[  275s] rootdir: /home/abuild/rpmbuild/BUILD/astropy-4.1rc1, inifile: setup.cfg, testpaths: astropy, docs
[  275s] plugins: hypothesis-5.11.0, astropy-header-0.1.2, asdf-2.6.0, mpl-0.11, arraydiff-0.3, filter-subpackage-0.0.0, openfiles-0.5.0, cov-2.10.0, doctestplus-0.7.0, remotedata-0.3.2
[  313s] collecting ... collected 15300 items / 2 skipped / 15298 selected
[  313s] 
...
[  895s] =================================== FAILURES ===================================
[  895s] ______________________________ test_spectral_cube ______________________________
[  895s] 
[  895s]     def test_spectral_cube():
[  895s]     
[  895s]         # Spectral cube with a weird axis ordering
[  895s]     
[  895s]         wcs = WCS_SPECTRAL_CUBE
[  895s]     
[  895s]         # Low-level API
[  895s]     
[  895s]         assert wcs.pixel_n_dim == 3
[  895s]         assert wcs.world_n_dim == 3
[  895s]         assert wcs.array_shape is None
[  895s]         assert wcs.pixel_shape is None
[  895s]         assert wcs.world_axis_physical_types == ['pos.galactic.lat', 'em.freq', 'pos.galactic.lon']
[  895s]         assert wcs.world_axis_units == ['deg', 'Hz', 'deg']
[  895s]         assert wcs.pixel_axis_names == ['', '', '']
[  895s]         assert wcs.world_axis_names == ['Latitude', 'Frequency', 'Longitude']
[  895s]     
[  895s]         assert_equal(wcs.axis_correlation_matrix, [[True, False, True],
[  895s]                                                    [False, True, False],
[  895s]                                                    [True, False, True]])
[  895s]     
[  895s]         assert len(wcs.world_axis_object_components) == 3
[  895s]         assert wcs.world_axis_object_components[0] == ('celestial', 1, 'spherical.lat.degree')
[  895s]         assert wcs.world_axis_object_components[1][:2] == ('spectral', 0)
[  895s]         assert wcs.world_axis_object_components[2] == ('celestial', 0, 'spherical.lon.degree')
[  895s]     
[  895s]         assert wcs.world_axis_object_classes['celestial'][0] is SkyCoord
[  895s]         assert wcs.world_axis_object_classes['celestial'][1] == ()
[  895s]         assert isinstance(wcs.world_axis_object_classes['celestial'][2]['frame'], Galactic)
[  895s]         assert wcs.world_axis_object_classes['celestial'][2]['unit'] is u.deg
[  895s]     
[  895s]         assert wcs.world_axis_object_classes['spectral'][0] is Quantity
[  895s]         assert wcs.world_axis_object_classes['spectral'][1] == ()
[  895s]         assert wcs.world_axis_object_classes['spectral'][2] == {}
[  895s]     
[  895s]         assert_allclose(wcs.pixel_to_world_values(29, 39, 44), (10, 20, 25))
[  895s]         assert_allclose(wcs.array_index_to_world_values(44, 39, 29), (10, 20, 25))
[  895s]     
[  895s]         assert_allclose(wcs.world_to_pixel_values(10, 20, 25), (29., 39., 44.))
[  895s]         assert_equal(wcs.world_to_array_index_values(10, 20, 25), (44, 39, 29))
[  895s]     
[  895s]         # High-level API
[  895s]     
[  895s]         coord, spec = wcs.pixel_to_world(29, 39, 44)
[  895s]         assert isinstance(coord, SkyCoord)
[  895s]         assert isinstance(coord.frame, Galactic)
[  895s] >       assert coord.l.deg == 25
[  895s] E       assert 24.999999999999993 == 25
[  895s] E         +24.999999999999993
[  895s] E         -25
[  895s] 
[  895s] astropy/wcs/wcsapi/tests/test_fitswcs.py:276: AssertionError
[  895s] =========================== short test summary info ============================
[  895s] FAILED astropy/wcs/wcsapi/tests/test_fitswcs.py::test_spectral_cube - assert ...
[  896s] ===== 1 failed, 14858 passed, 381 skipped, 62 xfailed in 620.59s (0:10:20) =====
```

Something returns a float where it should probably not. x86_64 does not fail on this test.

--- a/astropy/wcs/wcsapi/tests/test_fitswcs.py
+++ b/astropy/wcs/wcsapi/tests/test_fitswcs.py
@@ -151,14 +151,14 @@ def test_simple_celestial():
     coord = wcs.pixel_to_world(29, 39)
     assert isinstance(coord, SkyCoord)
     assert isinstance(coord.frame, ICRS)
-    assert coord.ra.deg == 10
-    assert coord.dec.deg == 20
+    assert_allclose(coord.ra.deg, 10)
+    assert_allclose(coord.dec.deg, 20)
 
     coord = wcs.array_index_to_world(39, 29)
     assert isinstance(coord, SkyCoord)
     assert isinstance(coord.frame, ICRS)
-    assert coord.ra.deg == 10
-    assert coord.dec.deg == 20
+    assert_allclose(coord.ra.deg, 10)
+    assert_allclose(coord.dec.deg, 20)
 
     coord = SkyCoord(10, 20, unit='deg', frame='icrs')
 
@@ -273,18 +273,18 @@ def test_spectral_cube():
     coord, spec = wcs.pixel_to_world(29, 39, 44)
     assert isinstance(coord, SkyCoord)
     assert isinstance(coord.frame, Galactic)
-    assert coord.l.deg == 25
-    assert coord.b.deg == 10
+    assert_allclose(coord.l.deg, 25)
+    assert_allclose(coord.b.deg, 10)
     assert isinstance(spec, SpectralCoord)
-    assert spec.to_value(u.Hz) == 20
+    assert_allclose(spec.to_value(u.Hz), 20)
 
     coord, spec = wcs.array_index_to_world(44, 39, 29)
     assert isinstance(coord, SkyCoord)
     assert isinstance(coord.frame, Galactic)
-    assert coord.l.deg == 25
-    assert coord.b.deg == 10
+    assert_allclose(coord.l.deg, 25)
+    assert_allclose(coord.b.deg, 10)
     assert isinstance(spec, SpectralCoord)
-    assert spec.to_value(u.Hz) == 20
+    assert_allclose(spec.to_value(u.Hz), 20)
 
     coord = SkyCoord(25, 10, unit='deg', frame='galactic')
     spec = 20 * u.Hz

